Rewrite Modding (#2948)
ppy/osu-wiki: wiki/Modding/en.md

# Modding

**Modding** is the act of providing feedback on another user's [beatmap](/wiki/Beatmaps). A user who is modding is referred to as a *modder*, and their collective posts on a beatmap's discussion are referred to as a *mod*.

Mods are mainly used to improve beatmaps aiming for Ranked status. Because it is a detailed and analytical process, modding beatmaps is also a popular way for users improve their understanding of beatmapping.

> ## Finding beatmaps to mod

Reviewer: this section should probably start off with a short paragraph saying there are many ways to find beatmaps to mod, and these are some of the main ones

jumping right into modding queues makes it seem like the whole section is going to be about modding queues

> [Modding Queues](https://osu.ppy.sh/community/forums/60) are the most actively utilized platform for finding beatmaps to mod and requesting mods. A user can create their own forum thread, referred to as a *queue*, for other users to post beatmaps that they want to be modded.

Reviewer: ```suggestion
The [Modding Queues forum](https://osu.ppy.sh/community/forums/60) is the most actively utilized platform for finding beatmaps to mod and requesting mods. A user can create their own thread, referred to as a *queue*, for other users to post beatmaps that they want to be modded.
```

> Each queue's creator dictates the queue's rules, controlling which beatmaps are viable for receiving mods. Queues are usually open for requests in two ways:

Reviewer: ```suggestion
Each queue's creator dictates the queue's rules, controlling which beatmaps are viable for receiving mods. Queues are usually open for requests of two kinds of mods:
```

- **Normal mod (NM):** A mod that will be provided without work from the requesting beatmapper.
- **Mod for mod (M4M):** A trade that requires each user to mod the other's beatmap.

> The `#modreqs` chat channel is populated by beatmappers posting maps they want modded. Most beatmaps posted do not receive mods.

Reviewer: probably dont mention this at all given how ineffective (and literally not functioning) it is

Well-known modders often note whether or not they are accepting mod requests on their user profiles.

Searching the pending beatmaps listing for beatmaps to mod is not recommended because most submissions are not aiming for Ranked status, so their creators will potentially not respond to mods.

> ## Modding process

Reviewer: something like "How to mod" is probably better cuz u speak to the reader directly in this part

> To mod a beatmapset, start by downloading the most up-to-date version and opening its discussion page. Play the difficulty you want to mod and keep in mind any gameplay-related issues. When completed, open the difficulty in the editor and analyze it for potential issues.

Reviewer: ```suggestion
To mod a beatmapset, start by downloading the most up-to-date version and opening its discussion page. Play the difficulty you want to mod and keep in mind any gameplay-related issues. When completed, open the difficulty in the editor and analyze it for other kinds of issues.
```

> When you come across an issue, select any relevant hit objects, then use the copy shortcut (`Ctrl` + `C`) to copy their timestamp information as text. On the beatmap's discussion page, locate the difficulty being modded, navigate to the applicable tab, paste (`Ctrl` + `V`) any applicable timestamps and your feedback. The `Timeline` tab is used for issues regarding specific parts of a beatmap, the `General (This difficulty)` tab is used for feedback on larger sections of a beatmap, and the `General (All difficulties)` tab is used for issues applicable to all difficulties on a beatmapset.

Reviewer: ```suggestion
When you come across an issue, select any relevant hit objects, then use the copy shortcut (`Ctrl` + `C`) to copy their timestamp information as text. On the beatmap's discussion, paste (`Ctrl` + `V`) any applicable timestamps and your feedback. The `Timeline` tab is used for issues regarding specific parts of a beatmap, the `General (This difficulty)` tab is used for feedback on larger sections of a beatmap, and the `General (All difficulties)` tab is used for issues applicable to all difficulties on a beatmapset.
```
obvious info is clarified a few sentences later anyway

> All mod posts should address what the issue is, why it is an issue, and how to fix it. For example, if a beatmap has an unnecessarily large jump, a mod could potentially be worded as "The spacing between these objects is too high. The song is not intense enough to justify this size of a jump, so moving the objects closer together would be more appropriate!". Many mod posts include links to images so that beatmappers can better understand any suggestions.

Reviewer: i feel like there's somewhere else to link for this (is it modding coc? i think? )

Modders should not assume a beatmapper understands why anything is an issue within their beatmap. When simplifying a mod post into a command like "Add a New Combo on this object", the beatmapper may not be convinced that the post is accurate and will likely deny the suggestion.

> Beatmappers regularly provide [kudosu](/wiki/Glossary#kudosu!) to mod posts that they agree with by clicking the *thumbs up* button. This statistic is tracked on an individual's profile and is referenced when applying to join the [Beatmap Nominators](/wiki/People/The_Team/Beatmap_Nominators).

Reviewer: also i think u shoudl refer to these as "votes" instead of thumbs up or w/e cuz theres two Buttons

> Beatmappers regularly provide [kudosu](/wiki/Glossary#kudosu!) to mod posts that they agree with by clicking the *thumbs up* button. This statistic is tracked on an individual's profile and is referenced when applying to join the [Beatmap Nominators](/wiki/People/The_Team/Beatmap_Nominators).

Author: added vote but kept thumbs up, since otherwise it's reads like thumbs down can give kudosu
